env.go

```diff
@@ -26,6 +29,12 @@ func ExtractEnv(secrets map[string]interface{}) (map[string]string, error) {
 	envSecrets := make(map[string]string, len(envMap))
 
 	for key, rawValue := range envMap {
+		// Reject keys that would be invalid environment variable identifiers
+		if !validIdentifierPattern.MatchString(key) {
+			err := fmt.Errorf("invalid identifier as key in environment: %q", key)
+
+			return nil, err
+		}
```

Comment:
According to the comment below, we seem to ignore when the values don't convert to strings properly, rather than return an error. IMO we should be loud about these failures (I think non-strings should also be errors, or at least print something to stderr), however if we really think it's better to silently continue, then we could change this to "only export keys that are valid identifiers" (and simply ignore invalid ones).

Separately, I may open a PR to add said stderr message (since making it an error would be a breaking change that could break production deployments).